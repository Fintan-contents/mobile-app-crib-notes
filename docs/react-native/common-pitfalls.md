---
title: ハマりがちな落とし穴
---

React Nativeで開発する際に起きた困り事や不具合についてまとめています。
