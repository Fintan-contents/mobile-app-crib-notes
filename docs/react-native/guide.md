---
title: 何らかのガイドを書きたい気持ちはある
---
