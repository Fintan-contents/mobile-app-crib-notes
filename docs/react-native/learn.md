---
title: About
hide_table_of_contents: true
---

このコンテンツはReact Nativeでのモバイルアプリケーション開発を学習するためのコンテンツです。

ReactでWebアプリケーションを実装できるがReact Nativeでモバイルアプリケーションを実装したことがないアプリケーション開発者を対象として、次のことについて学ぶことを目的としています。

- ReactとReact Nativeの違い
- React Nativeで作成したモバイルアプリケーションの動作確認方法

また、作成したモバイルアプリケーションの配布方法やテスト手法についてはこのコンテンツでは触れません。

<!-- textlint-disable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Getting Started',
    to: '/react-native/learn/getting-started',
    summary: <>React Nativeでのアプリ開発に必要な環境を用意します。</>
  },
  {
    title: 'アプリの実装前に',
    to: '/react-native/learn/basic-concepts',
    summary: <>React Nativeでアプリを実装し始める前に最低限必要となる前提知識や、React NativeとReact Navigationの基本的な知識について学びます。</>
  },
  {
    title: 'ToDoアプリの実装',
    to: '/react-native/learn/todo-app',
    summary: <>React Nativeで簡単なToDoアプリを実装します。</>
  },
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->
