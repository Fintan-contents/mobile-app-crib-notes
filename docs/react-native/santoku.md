---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

React Nativeサンプルアプリケーションの設計ドキュメントです。

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
