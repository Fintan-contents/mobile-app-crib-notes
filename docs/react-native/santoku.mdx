---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Development',
    summary: '開発環境やビルドに関するドキュメント',
    to: '/react-native/santoku/development',
  }
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
