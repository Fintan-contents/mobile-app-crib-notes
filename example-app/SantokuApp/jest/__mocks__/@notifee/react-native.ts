import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
