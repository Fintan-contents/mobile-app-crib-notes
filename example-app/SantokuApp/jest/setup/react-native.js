// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
