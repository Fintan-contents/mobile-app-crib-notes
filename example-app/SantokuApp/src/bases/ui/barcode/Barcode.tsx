// Original source:
//   https://github.com/Kichiyaki/react-native-barcode-generator/blob/22f020eacf47fe23f0e311732ec801068db14fbc/index.js
// Original license:
//   The MIT License (MIT)
//
//   Copyright (c) 2021 Dawid Wysokiński
//
//   Permission is hereby granted, free of charge, to any person obtaining a copy
//   of this software and associated documentation files (the "Software"), to deal
//   in the Software without restriction, including without limitation the rights
//   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
//   copies of the Software, and to permit persons to whom the Software is
//   furnished to do so, subject to the following conditions:
//
//   The above copyright notice and this permission notice shall be included in all
//   copies or substantial portions of the Software.
//
//   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
//   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
//   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
//   SOFTWARE.

import CODE128 from 'jsbarcode/src/barcodes/CODE128/CODE128';
import CODE128AUTO from 'jsbarcode/src/barcodes/CODE128/CODE128_AUTO';
import React, {useCallback, useMemo} from 'react';
import {StyleSheet, Text, TextProps, View, ViewProps} from 'react-native';
import {Path, Svg} from 'react-native-svg';

export type Format = 'CODE128' | 'CODE128AUTO';
export const START_CHARACTERS = {
  'CODE-A': String.fromCharCode(105 + 103),
  'CODE-B': String.fromCharCode(105 + 104),
  'CODE-C': String.fromCharCode(105 + 105),
};

export const CODE_SET_CHARACTERS = {
  'CODE-A': String.fromCharCode(105 + 101),
  'CODE-B': String.fromCharCode(105 + 100),
  'CODE-C': String.fromCharCode(105 + 99),
};
const ENCODERS = {
  CODE128,
  CODE128AUTO,
};

export type BarcodeProps = {
  /**
   * バーコードに設定するデータ
   */
  value: string;
  /**
   * バーの幅
   */
  lineWidth?: number;
  /**
   * バーコードの最大幅（textで表示する領域は含まない）
   */
  maxWidth?: number;
  /**
   * バーコードの高さ（textで表示する領域は含まない）
   */
  height?: number;
  /**
   * バーコードの左右の余白
   */
  quietZone?: number;
  /**
   * バーコードのフォーマット
   * 以下のフォーマットを設定可能
   * - CODE128: スタートキャラクタやコードセットキャラクタを自身で設定するフォーマット
   * - CODE128AUTO: 指定されたデータに応じて自動でスタートキャラクタやコードセットキャラクタを設定するフォーマット
   */
  format?: Format;
  /**
   * バーの色
   */
  lineColor?: string;
  /**
   * 背景色
   */
  background?: string;
  /**
   * バーコード下部に表示するテキスト
   */
  text?: string;
  /**
   * バーコード下部に表示するテキストのProps
   */
  textProps?: TextProps;
  /**
   * ViewProps
   */
  viewProps?: ViewProps;
  /**
   * エラー発生時に呼び出されるコールバック関数
   */
  onError?: (err: any) => void;
};

const initialBarcode = {
  bars: '',
  barCodeWidth: 0,
};

const drawRect = (x: number, y: number, width: number, height: number) => {
  return `M${x},${y}h${width}v${height}h-${width}z`;
};

const drawSvgBarCode = (binary: string, height: number, barCodeWidth: number) => {
  const rects = [];

  const singleBarWidth = barCodeWidth / binary.length;
  let barWidth = 0;
  let x = 0;
  const yFrom = 0;

  for (let b = 0; b < binary.length; b++) {
    x = b * singleBarWidth;
    if (binary[b] === '1') {
      barWidth++;
    } else if (barWidth > 0) {
      rects[rects.length] = drawRect(x - singleBarWidth * barWidth, yFrom, singleBarWidth * barWidth, height);
      barWidth = 0;
    }
  }

  if (barWidth > 0) {
    rects[rects.length] = drawRect(x - singleBarWidth * (barWidth - 1), yFrom, singleBarWidth * barWidth, height);
  }

  return rects;
};

/**
 * バーコードを生成し、表示するコンポーネント
 * CODE128のバーコードのみをサポートしている
 */
export const Barcode: React.FC<BarcodeProps> = ({
  value = '',
  lineWidth = 2,
  maxWidth = Infinity,
  height = 100,
  quietZone = 0,
  format = 'CODE128',
  lineColor = '#000000',
  background = '#FFFFFF',
  text,
  textProps: {style: textStyle, ...textProps} = {},
  viewProps: {style, ...viewProps} = {},
  onError,
}) => {
  const getEncoder = useCallback(() => {
    const encoder = new ENCODERS[format](value, {});
    if (!encoder.valid()) {
      throw new Error('Invalid barcode for selected format.');
    }
    return encoder;
  }, [format, value]);

  const {bars, barCodeWidth} = useMemo(() => {
    try {
      const encoder = getEncoder();
      const encoded = encoder.encode();
      const barCodeWidth = Math.min(maxWidth, encoded.data.length * lineWidth);
      return {
        bars: drawSvgBarCode(encoded.data, height, barCodeWidth).join(' '),
        barCodeWidth,
      };
    } catch (error) {
      onError?.(error);
    }
    return initialBarcode;
  }, [getEncoder, lineWidth, height, maxWidth, onError]);

  return (
    <View
      style={StyleSheet.flatten([styles.container, {backgroundColor: background, paddingHorizontal: quietZone}, style])}
      {...viewProps}>
      <Svg height={height} width={barCodeWidth} fill={lineColor}>
        <Path d={bars} />
      </Svg>
      {text && (
        <Text style={StyleSheet.flatten([styles.textStyle, textStyle])} {...textProps}>
          {text}
        </Text>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  textStyle: {
    textAlign: 'center',
  },
});
