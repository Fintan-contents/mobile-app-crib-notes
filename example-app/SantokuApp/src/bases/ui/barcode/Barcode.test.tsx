import {render, screen} from '@testing-library/react-native';
import React, {SVGProps} from 'react';
import {TextProps, ViewProps} from 'react-native';
import {PathProps} from 'react-native-svg';
import {ReactTestInstance} from 'react-test-renderer';

import {Barcode, CODE_SET_CHARACTERS, START_CHARACTERS} from './Barcode';

const getSvgProps = (view: ReactTestInstance): SVGProps<any> => {
  const svgView = view.children[0] as ReactTestInstance;
  return svgView.props;
};

const getSvgPathProps = (view: ReactTestInstance): PathProps => {
  const svgView = view.children[0] as ReactTestInstance;
  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- svgView.props.childrenがany型になってしまうため、型を強制的に指定
  return svgView.props.children.props as PathProps;
};

describe('Barcode only with required props', () => {
  it('renders successfully only with required props', () => {
    render(
      <Barcode
        value={`${START_CHARACTERS['CODE-A']}0123456789${CODE_SET_CHARACTERS['CODE-C']}012345`}
        viewProps={{testID: 'Barcode'}}
      />,
    );
    const barcode = screen.queryByTestId('Barcode');
    expect(barcode).not.toBeNull();
    expect(screen).toMatchSnapshot();
  });
});

describe('Barcode with all format', () => {
  it('renders successfully with CODE128', () => {
    render(
      <Barcode
        value={`${START_CHARACTERS['CODE-A']}0123456789${CODE_SET_CHARACTERS['CODE-C']}012345`}
        format="CODE128"
        viewProps={{testID: 'Barcode'}}
      />,
    );
    expect(screen).toMatchSnapshot();
    const view = screen.getByTestId('Barcode');
    const svgPathProps = getSvgPathProps(view);
    expect(svgPathProps.d).toBe(
      'M0,0h4v100h-4z M6,0h2v100h-2z M16,0h2v100h-2z M22,0h2v100h-2z M28,0h6v100h-6z M36,0h4v100h-4z M44,0h2v100h-2z M50,0h6v100h-6z M60,0h4v100h-4z M66,0h4v100h-4z M74,0h6v100h-6z M84,0h2v100h-2z M88,0h4v100h-4z M96,0h2v100h-2z M100,0h6v100h-6z M110,0h4v100h-4z M118,0h2v100h-2z M124,0h6v100h-6z M132,0h4v100h-4z M138,0h6v100h-6z M148,0h2v100h-2z M154,0h4v100h-4z M162,0h6v100h-6z M170,0h2v100h-2z M176,0h6v100h-6z M184,0h4v100h-4z M190,0h6v100h-6z M198,0h6v100h-6z M206,0h2v100h-2z M212,0h4v100h-4z M220,0h6v100h-6z M230,0h2v100h-2z M234,0h4v100h-4z M242,0h2v100h-2z M246,0h6v100h-6z M254,0h8v100h-8z M264,0h4v100h-4z M272,0h4v100h-4z M278,0h4v100h-4z M286,0h6v100h-6z M294,0h4v100h-4z M300,0h6v100h-6z M308,0h2v100h-2z M312,0h6v100h-6z M320,0h4v100h-4z M330,0h2v100h-2z M338,0h6v100h-6z M346,0h4v100h-4z M352,0h4v100h-4z M362,0h6v100h-6z M370,0h2v100h-2z M374,0h4v100h-4z',
    );
  });
  it('renders successfully with CODE128AUTO', () => {
    render(<Barcode value="0123456789" format="CODE128AUTO" viewProps={{testID: 'Barcode'}} />);
    expect(screen).toMatchSnapshot();
    const view = screen.getByTestId('Barcode');
    const svgPathProps = getSvgPathProps(view);
    expect(svgPathProps.d).toBe(
      'M0,0h4v100h-4z M6,0h2v100h-2z M12,0h6v100h-6z M22,0h4v100h-4z M30,0h4v100h-4z M36,0h4v100h-4z M44,0h6v100h-6z M52,0h4v100h-4z M58,0h6v100h-6z M66,0h2v100h-2z M70,0h6v100h-6z M78,0h4v100h-4z M88,0h2v100h-2z M98,0h2v100h-2z M102,0h4v100h-4z M110,0h4v100h-4z M116,0h4v100h-4z M122,0h8v100h-8z M132,0h2v100h-2z M142,0h4v100h-4z M148,0h2v100h-2z M154,0h4v100h-4z M164,0h6v100h-6z M172,0h2v100h-2z M176,0h4v100h-4z',
    );
  });
});

describe('Barcode with onError', () => {
  it('should be called on error', () => {
    const onError = jest.fn();
    // CODE128では使用できない値を設定
    render(<Barcode value="あいうえお" format="CODE128" onError={onError} viewProps={{testID: 'Barcode'}} />);
    expect(onError).toHaveBeenCalledTimes(1);
  });
});

describe('Barcode with all props', () => {
  it('should be applied properly', () => {
    render(
      <Barcode
        value="0123456789"
        format="CODE128AUTO"
        lineWidth={5}
        maxWidth={300}
        height={200}
        quietZone={5}
        lineColor="red"
        background="blue"
        text="012345"
        viewProps={{testID: 'Barcode', style: {alignItems: 'flex-start'}}}
        textProps={{testID: 'BarcodeText', style: {textAlign: 'left', color: 'yellow'}}}
      />,
    );
    expect(screen).toMatchSnapshot();
    const view = screen.getByTestId('Barcode');
    const viewProps = view.props as ViewProps;
    expect(viewProps.style).toEqual({
      backgroundColor: 'blue',
      alignItems: 'flex-start',
      paddingHorizontal: 5,
    });
    const svgProps = getSvgProps(view);
    const svgPathProps = getSvgPathProps(view);
    expect(svgProps.fill).toBe('red');
    expect(svgProps.height).toBe(200);
    expect(svgPathProps.d).toBe(
      'M0,0h6.666666666666667v200h-6.666666666666667z M10,0h3.3333333333333335v200h-3.3333333333333335z M20,0h10v200h-10z M36.66666666666667,0h6.666666666666667v200h-6.666666666666667z M50.00000000000001,0h6.666666666666667v200h-6.666666666666667z M60.00000000000001,0h6.666666666666667v200h-6.666666666666667z M73.33333333333334,0h10v200h-10z M86.66666666666667,0h6.666666666666667v200h-6.666666666666667z M96.66666666666667,0h10v200h-10z M110.00000000000001,0h3.3333333333333335v200h-3.3333333333333335z M116.66666666666667,0h10v200h-10z M130.00000000000003,0h6.666666666666667v200h-6.666666666666667z M146.66666666666666,0h3.3333333333333335v200h-3.3333333333333335z M163.33333333333334,0h3.3333333333333335v200h-3.3333333333333335z M170.00000000000003,0h6.666666666666667v200h-6.666666666666667z M183.33333333333334,0h6.666666666666667v200h-6.666666666666667z M193.33333333333334,0h6.666666666666667v200h-6.666666666666667z M203.33333333333334,0h13.333333333333334v200h-13.333333333333334z M220,0h3.3333333333333335v200h-3.3333333333333335z M236.66666666666669,0h6.666666666666667v200h-6.666666666666667z M246.66666666666666,0h3.3333333333333335v200h-3.3333333333333335z M256.6666666666667,0h6.666666666666667v200h-6.666666666666667z M273.33333333333337,0h10v200h-10z M286.6666666666667,0h3.3333333333333335v200h-3.3333333333333335z M293.33333333333337,0h6.666666666666667v200h-6.666666666666667z',
    );

    const text = screen.getByTestId('BarcodeText');
    const textProps = text.props as TextProps;
    expect(textProps.style).toEqual({textAlign: 'left', color: 'yellow'});
    const textValue = screen.queryByText('012345');
    expect(textValue).not.toBeNull();
  });
});
