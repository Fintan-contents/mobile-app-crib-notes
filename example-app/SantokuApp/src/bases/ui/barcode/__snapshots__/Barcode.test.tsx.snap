// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Barcode only with required props renders successfully only with required props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "paddingHorizontal": 0,
    }
  }
  testID="Barcode"
>
  <RNSVGSvgView
    bbHeight="100"
    bbWidth="378"
    fill="#000000"
    focusable={false}
    height={100}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 378,
        },
      ]
    }
    width={378}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0,0h4v100h-4z M6,0h2v100h-2z M16,0h2v100h-2z M22,0h2v100h-2z M28,0h6v100h-6z M36,0h4v100h-4z M44,0h2v100h-2z M50,0h6v100h-6z M60,0h4v100h-4z M66,0h4v100h-4z M74,0h6v100h-6z M84,0h2v100h-2z M88,0h4v100h-4z M96,0h2v100h-2z M100,0h6v100h-6z M110,0h4v100h-4z M118,0h2v100h-2z M124,0h6v100h-6z M132,0h4v100h-4z M138,0h6v100h-6z M148,0h2v100h-2z M154,0h4v100h-4z M162,0h6v100h-6z M170,0h2v100h-2z M176,0h6v100h-6z M184,0h4v100h-4z M190,0h6v100h-6z M198,0h6v100h-6z M206,0h2v100h-2z M212,0h4v100h-4z M220,0h6v100h-6z M230,0h2v100h-2z M234,0h4v100h-4z M242,0h2v100h-2z M246,0h6v100h-6z M254,0h8v100h-8z M264,0h4v100h-4z M272,0h4v100h-4z M278,0h4v100h-4z M286,0h6v100h-6z M294,0h4v100h-4z M300,0h6v100h-6z M308,0h2v100h-2z M312,0h6v100h-6z M320,0h4v100h-4z M330,0h2v100h-2z M338,0h6v100h-6z M346,0h4v100h-4z M352,0h4v100h-4z M362,0h6v100h-6z M370,0h2v100h-2z M374,0h4v100h-4z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Barcode with all format renders successfully with CODE128 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "paddingHorizontal": 0,
    }
  }
  testID="Barcode"
>
  <RNSVGSvgView
    bbHeight="100"
    bbWidth="378"
    fill="#000000"
    focusable={false}
    height={100}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 378,
        },
      ]
    }
    width={378}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0,0h4v100h-4z M6,0h2v100h-2z M16,0h2v100h-2z M22,0h2v100h-2z M28,0h6v100h-6z M36,0h4v100h-4z M44,0h2v100h-2z M50,0h6v100h-6z M60,0h4v100h-4z M66,0h4v100h-4z M74,0h6v100h-6z M84,0h2v100h-2z M88,0h4v100h-4z M96,0h2v100h-2z M100,0h6v100h-6z M110,0h4v100h-4z M118,0h2v100h-2z M124,0h6v100h-6z M132,0h4v100h-4z M138,0h6v100h-6z M148,0h2v100h-2z M154,0h4v100h-4z M162,0h6v100h-6z M170,0h2v100h-2z M176,0h6v100h-6z M184,0h4v100h-4z M190,0h6v100h-6z M198,0h6v100h-6z M206,0h2v100h-2z M212,0h4v100h-4z M220,0h6v100h-6z M230,0h2v100h-2z M234,0h4v100h-4z M242,0h2v100h-2z M246,0h6v100h-6z M254,0h8v100h-8z M264,0h4v100h-4z M272,0h4v100h-4z M278,0h4v100h-4z M286,0h6v100h-6z M294,0h4v100h-4z M300,0h6v100h-6z M308,0h2v100h-2z M312,0h6v100h-6z M320,0h4v100h-4z M330,0h2v100h-2z M338,0h6v100h-6z M346,0h4v100h-4z M352,0h4v100h-4z M362,0h6v100h-6z M370,0h2v100h-2z M374,0h4v100h-4z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Barcode with all format renders successfully with CODE128AUTO 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "paddingHorizontal": 0,
    }
  }
  testID="Barcode"
>
  <RNSVGSvgView
    bbHeight="100"
    bbWidth="180"
    fill="#000000"
    focusable={false}
    height={100}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 180,
        },
      ]
    }
    width={180}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0,0h4v100h-4z M6,0h2v100h-2z M12,0h6v100h-6z M22,0h4v100h-4z M30,0h4v100h-4z M36,0h4v100h-4z M44,0h6v100h-6z M52,0h4v100h-4z M58,0h6v100h-6z M66,0h2v100h-2z M70,0h6v100h-6z M78,0h4v100h-4z M88,0h2v100h-2z M98,0h2v100h-2z M102,0h4v100h-4z M110,0h4v100h-4z M116,0h4v100h-4z M122,0h8v100h-8z M132,0h2v100h-2z M142,0h4v100h-4z M148,0h2v100h-2z M154,0h4v100h-4z M164,0h6v100h-6z M172,0h2v100h-2z M176,0h4v100h-4z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Barcode with all props should be applied properly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "blue",
      "paddingHorizontal": 5,
    }
  }
  testID="Barcode"
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="300"
    fill="red"
    focusable={false}
    height={200}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 200,
          "width": 300,
        },
      ]
    }
    width={300}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294901760,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0,0h6.666666666666667v200h-6.666666666666667z M10,0h3.3333333333333335v200h-3.3333333333333335z M20,0h10v200h-10z M36.66666666666667,0h6.666666666666667v200h-6.666666666666667z M50.00000000000001,0h6.666666666666667v200h-6.666666666666667z M60.00000000000001,0h6.666666666666667v200h-6.666666666666667z M73.33333333333334,0h10v200h-10z M86.66666666666667,0h6.666666666666667v200h-6.666666666666667z M96.66666666666667,0h10v200h-10z M110.00000000000001,0h3.3333333333333335v200h-3.3333333333333335z M116.66666666666667,0h10v200h-10z M130.00000000000003,0h6.666666666666667v200h-6.666666666666667z M146.66666666666666,0h3.3333333333333335v200h-3.3333333333333335z M163.33333333333334,0h3.3333333333333335v200h-3.3333333333333335z M170.00000000000003,0h6.666666666666667v200h-6.666666666666667z M183.33333333333334,0h6.666666666666667v200h-6.666666666666667z M193.33333333333334,0h6.666666666666667v200h-6.666666666666667z M203.33333333333334,0h13.333333333333334v200h-13.333333333333334z M220,0h3.3333333333333335v200h-3.3333333333333335z M236.66666666666669,0h6.666666666666667v200h-6.666666666666667z M246.66666666666666,0h3.3333333333333335v200h-3.3333333333333335z M256.6666666666667,0h6.666666666666667v200h-6.666666666666667z M273.33333333333337,0h10v200h-10z M286.6666666666667,0h3.3333333333333335v200h-3.3333333333333335z M293.33333333333337,0h6.666666666666667v200h-6.666666666666667z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "yellow",
        "textAlign": "left",
      }
    }
    testID="BarcodeText"
  >
    012345
  </Text>
</View>
`;
